strawberryfields/backends/fockbackend/circuit.py

```diff
@@ -335,6 +343,37 @@ def _apply_BS(mat, state, trunc):  # pragma: no cover
                     ret[i, j] += mat[i, k, j, i + j - k] * state[k, i + j - k]
         return ret
 
+    # ignored in covtest (doesn't work well with the jit decorator)
+    @staticmethod
+    @jit(nopython=True)
+    def _apply_MZ(mat, state, trunc):  # pragma: no cover
+        r"""Applies the MZ gate to the first bra in state.
+
+        The Mach-Zehnder matrix elements :math:`B_{ij}^{kl}` satisfy the selection
+        rules
+
+        .. math::
+            MZ_{ij}^{kl} \propto \delta_{i+j}^{k+l}
+
+        This implies that one can contract pure states (without loss of generality
+        we assume only two modes) using one less loop as follows:
+
+        .. math::
+            c'_{i, j} = \sum\limits_{k = \text{max}(1+i+j-n, 0)}
+                    ^{\text{min}(i+j, n-1) + 1}
+            B_{ik}^{j, i+j-k} c_{k, i+j-k}
+
+        where :math:`c_{kl}` is the tensor representing the two mode state and
+        :math:`n` is the cutoff in Fock space. Similarly for a mixed state it is
+        possible to do the update of the state using two less summations.
+        """
```

Comment:
wondering if this function is needed since it does exactly the same as the `applyBS` one. Maybe you can delete this one, rename the previous one `apply_two_mode_passive` and update the functions calls in BS and MZ